lib/grape/dsl/parameters.rb

```diff
@@ -191,7 +191,9 @@ def params(params)
         params = @parent.params(params) if @parent
         if @element
           params = if params.is_a?(Array)
-                     params.flat_map { |el| el[@element] || {} }
+                     # we can get nested arrays as result
+                     # it used for calculating parent array indicies for error messages
+                     params.map { |el| el[@element] || {} }
```

Comment:
I think you should just leave the second part of the comment, and correct spelling, ie. `used for calculating parent array indices for error messages`